From f57b85f3ab9feaf74d9af30fe315f9bf6b5b3ae0 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:28:45 -0400
Subject: [acpi] bind: Redefine acpi_get_child()

Message-id: <1411129776-21656-11-git-send-email-prarit@redhat.com>
Patchwork-id: 96349
O-Subject: [RHEL7.1 PATCH BZ 1141846 10/61] ACPI / bind: Redefine acpi_get_child()
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit 11dcc75dba5bf8b69c4612de10e366c4e04cb123
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Thu Nov 28 23:58:18 2013 +0100

    ACPI / bind: Redefine acpi_get_child()

    Since acpi_get_child() is the only user of acpi_find_child() now,
    drop the static inline definition of the former and redefine the
    latter as new acpi_get_child().

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Reviewed-by: Aaron Lu <aaron.lu@intel.com>
    Tested-by: Aaron Lu <aaron.lu@intel.com> # for ATA binding

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Stefan Assmann <sassmann@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/glue.c b/drivers/acpi/glue.c
index d854492..29a3fb1 100644
--- a/drivers/acpi/glue.c
+++ b/drivers/acpi/glue.c
@@ -150,17 +150,17 @@ struct acpi_device *acpi_find_child_device(struct acpi_device *parent,
  return ret;
 }
 
-acpi_handle acpi_find_child(acpi_handle handle, u64 addr, bool is_bridge)
+acpi_handle acpi_get_child(acpi_handle handle, u64 addr)
 {
  struct acpi_device *adev;
 
  if (!handle || acpi_bus_get_device(handle, &adev))
   return NULL;
 
- adev = acpi_find_child_device(adev, addr, is_bridge);
+ adev = acpi_find_child_device(adev, addr, false);
  return adev ? adev->handle : NULL;
 }
-EXPORT_SYMBOL_GPL(acpi_find_child);
+EXPORT_SYMBOL_GPL(acpi_get_child);
 
 static void acpi_physnode_link_name(char *buf, unsigned int node_id)
 {
diff --git a/include/acpi/acpi_bus.h b/include/acpi/acpi_bus.h
index e8f0f2b..3e3d716 100644
--- a/include/acpi/acpi_bus.h
+++ b/include/acpi/acpi_bus.h
@@ -443,11 +443,7 @@ struct acpi_pci_root {
 
 struct acpi_device *acpi_find_child_device(struct acpi_device *parent,
         u64 address, bool check_children);
-acpi_handle acpi_find_child(acpi_handle, u64, bool);
-static inline acpi_handle acpi_get_child(acpi_handle handle, u64 addr)
-{
- return acpi_find_child(handle, addr, false);
-}
+acpi_handle acpi_get_child(acpi_handle handle, u64 addr);
 void acpi_preset_companion(struct device *dev, acpi_handle parent, u64 addr);
 int acpi_is_root_bridge(acpi_handle);
 struct acpi_pci_root *acpi_pci_find_root(acpi_handle handle);
-- 
1.7.1